### PR TITLE
Add method to check capability to access to the top-level WooCommerce menu item.

### PR DESCRIPTION
**Pull request by octaedro**
This PR is part of this fix (19-gh-woocommerce/woocommerce-admin-ghsa-qxgw-7667-cprq).
The method `can_view_woocommerce_menu_item` is added to check if the user can access the top-level WooCommerce menu item.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes one of [these issues](https://github.com/woocommerce/woocommerce-admin/security/advisories/GHSA-qxgw-7667-cprq). More context here (p90Yrv-2vK-p2#comment-6479).

The method added in this PR (`can_view_woocommerce_menu_item`) will be used inside the method `is_admin_user` (WC-Admin) to check if a user can `manage_woocommerce` or `edit_others_shop_orders`. I'm creating this method in `core` to prevent getting out of sync with WC-Admin.

### How to test the changes in this Pull Request:

Follow the steps described in this PR (19-gh-woocommerce/woocommerce-admin-ghsa-qxgw-7667-cprq).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Add method `can_view_woocommerce_menu_item` to check if the user can access the top-level WooCommerce menu item.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/admin/class-wc-admin-menus.php
Comment (joshuatf) [inline plugins/woocommerce/includes/admin/class-wc-admin-menus.php:93]: Nice!  Looks good overall, but before we merge this in, I'm also wondering if a filter would work for our needs here.

```php
return apply_filters( 'woocommerce_menu_item_visibility', current_user_can( 'edit_others_shop_orders' ) );
```

Does this work for the use case in WooCommerce Admin or is a separate menu item altogether?

**Review by joshuatf — COMMENTED** [review]
Thanks for tackling this, @octaedro!

There are also a few spots in this file using `current_user_can( 'edit_others_shop_orders' )` that could benefit from calling `self::can_view_woocommerce_menu_item()`.

File: plugins/woocommerce/includes/admin/class-wc-admin-menus.php
Comment (octaedro) [reply plugins/woocommerce/includes/admin/class-wc-admin-menus.php:93]: ~Good call! I added the filter in commit 2bc2d54~

> ~Does this work for the use case in WooCommerce Admin or is a separate menu item altogether?~

~I like the hook addition and it looks like it'll still work for our use case in WooCommerce Admin.~

**Comment by octaedro** [comment]
> There are also a few spots in this file using current_user_can( 'edit_others_shop_orders' ) that could benefit from calling self::can_view_woocommerce_menu_item().

Good idea! I added these changes in commit 8e1bd70. I didn't modify [this](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/class-wc-admin-menus.php#L188) and [this](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/class-wc-admin-menus.php#L244) because in those cases it seems more relevant to know if the user has that capability than if they `can_view_woocommerce_menu_item`.

@joshuatf Could you take another look at this PR?

File: plugins/woocommerce/includes/admin/class-wc-admin-menus.php
Comment (octaedro) [reply plugins/woocommerce/includes/admin/class-wc-admin-menus.php:93]: After analyzing in detail with @joshuatf the addition of this filter, we decided to roll this change back.

**Review by joshuatf — DISMISSED** [review]
Thanks for the great discussion on this, Fernando!

This looks good and is testing well.  The only other place I can see in this file that might be worth using the check is in `custom_menu_order`, but pre-approving and will leave up to your discretion.

**Comment by octaedro** [comment]
Thank you @joshuatf for your review.

> The only other place I can see in this file that might be worth using the check is in custom_menu_order

Sounds good to me. I add it in commit c065121. It seems like the new commit dismissed your previous review 🙁

Could you take another look at this PR?

**Review by joshuatf — APPROVED** [review]
LGTM!  Thanks for the change!

**Pull request merged**
